### DISCUSSION TITLE
How can I get uid of all documents if I use useFirestoreCollectionData hook ?

### DISCUSSION THREAD

**Discussion by catrawalkar** [FirebaseExtended/reactfire#257] [Q&A] [answered]
### Version info

**React:** 

**Firebase:**

**ReactFire:**

**Other (e.g. Node, browser, operating system) (if applicable):**

### Test case

After fetching all the documents from a collection, if i want to update one document, how can I do it? I think it will require document uid for this. There is no way to get document ids of all documents.

Is is possible to pass a flag to useFirestoreCollection and useFirestoreCollectionData which will do this:
```
snapshot.forEach(data=>{
    return {uid: data.uid, ...data.data()}
})
```

**Comment by jhuleatt** [comment, marked as answer]
Hi @catrawalkar , `useFirestoreCollectionData` accepts a config object that allows you to add an id field to each object in the array it returns. The sample shows how this is done:

https://github.com/FirebaseExtended/reactfire/blob/68cfc7957caa317322fc745ad0880b34a4df1065/sample/src/Firestore.js#L74

This will add an `id` key to every result. The returned objects will look like this:

```js
[
    {
        commonName: 'Giraffe',
        id: 'theDocumentId' // the key is `id` because we set `idField` to "id"
    }
]
```

It looks like the [reference](https://github.com/FirebaseExtended/reactfire/blob/master/docs/reference.md#useFirestoreCollectionData) is missing this information. Sorry about that! Any chance you have a moment to create a PR for that? Otherwise I can add it :)

**Comment by catrawalkar** [comment]
@jhuleatt  Sure, I will try to create a PR for this. What other things can be configured using config object?

Also, before opening an issue, i tried a lot to figure out if this is possible by digging out in the code. I was not able to figure out due to lack of RxJS, TS know how..! Can you help me with good resources to learn these things so I can contribute here?

**Comment by jhuleatt** [comment]
Great!

Re: learning: It might help if I explain the source of `useFirestoreCollectionData` a bit:

https://github.com/FirebaseExtended/reactfire/blob/68cfc7957caa317322fc745ad0880b34a4df1065/reactfire/firestore/index.tsx#L111-L116

The `options` argument's type is `ReactFireOptions`:

https://github.com/FirebaseExtended/reactfire/blob/68cfc7957caa317322fc745ad0880b34a4df1065/reactfire/index.ts#L1-L4

It uses RxFire's [`collectionData`](https://github.com/FirebaseExtended/reactfire/blob/68cfc7957caa317322fc745ad0880b34a4df1065/reactfire/index.ts#L1-L4) observable to subscribe to a collection and extract an object from each document
